.circleci/config.yml

```diff
@@ -12,76 +12,183 @@ version: 2.1
+            git config credential.helper 'cache --timeout=<< parameters.cache_timeout >>'
+            git config user.email "<< parameters.gh_email >>"
+            git config user.name "<< parameters.gh_username >>"
+
+  gus-prepare-environment-variables:
+    parameters:
+      package-json-path:
+        type: string
+        default: './packages/salesforcedx-vscode/package.json'
+      release-tag-name:
+        type: string
+        default: 'offcore.tooling'
+    steps:
+      - run:
+          name: 'Prepare environment variables for GUS Change Case Management'
+          command: |
+            export GUS_BUILD="$(node -pe "require('<< parameters.package-json-path >>').version")"
+            echo "export RELEASE_VERSION=$GUS_BUILD" >> $BASH_ENV
+            echo "export GUS_RELEASE=<< parameters.release-tag-name >>.$GUS_BUILD" >> $BASH_ENV
+            source $BASH_ENV
+            echo "RELEASE_VERSION is: ${RELEASE_VERSION}"
+            echo "GUS_RELEASE is: ${GUS_RELEASE}"
+
+  install:
+    description: 'Installing dependencies'
+    steps:
+      - run: npm install
+      - run: npm --version
+      - run: node --version
+
+  package-extensions:
+    steps:
       - run:
           name: 'Packaging extensions'
           command: npm run vscode:package
+
+  restore_dependency_cache:
+    # This logic cannot be done within the node orb currently. This is due to the fact that we use
+    # the package.json instead of the package-lock.json for our checksum.
+    parameters:
+      base-cache-path:
+        type: string
+        default: 'dependency-cache-v2-'
+    steps:
+      - restore_cache:
+          keys:
+            - << parameters.base-cache-path >>{{ checksum "package.json" }}
+            # if cache for exact version of 'package.json' is not present, then load the most recent cache
+            - << parameters.base-cache-path >>
+
+  save_dependency_cache:
+    parameters:
+      base-cache-path:
+        type: string
+        default: 'dependency-cache-v2-'
+      path-to-save:
+        type: string
+        default: 'node_modules'
+    steps:
+      - save_cache:
+          key: << parameters.base-cache-path >>{{ checksum "package.json" }}
+          paths:
+            - << parameters.path-to-save >>
+
+  ssh-config:
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - << pipeline.parameters.fingerprint >>
+
+  stage-and-store-artifacts:
+    parameters:
+      location:
+        type: string
+        default: 'extensions'
+    steps:
       - run:
           name: 'Staging artifacts'
           command: |
-            mkdir extensions
-            find packages -name "*.vsix" -type f -exec cp {} ./extensions \;
+            mkdir << parameters.location >>
+            find packages -name "*.vsix" -type f -exec cp {} ./<< parameters.location >> \;
       - store_artifacts:
-          path: ./extensions
-      - save_cache:
-          key: dependency-cache-v2-{{ checksum "package.json" }}
```

Comment:
Just a pointer - caching should be better now for the build-all step. We weren't using the same key when restoring the cache as we were for saving in the build-artifacts command. So we were always rebuilding the cache for every build-all job.

Reply:
Good catch 👍